bindings/cs/rl.net.native/rl.net.live_model.cc

```diff
@@ -1,33 +1,56 @@
 #include "rl.net.live_model.h"
+#include "binding_tracer.h"
+#include "constants.h"
+#include "err_constants.h"
+#include "rl.net.live_model.h"
+#include "trace_logger.h"
 
-static void pipe_managed_callback(const reinforcement_learning::api_status& status, livemodel_context_t* context)
+
+static void pipe_backgroud_error_callback(const reinforcement_learning::api_status& status, livemodel_context_t* context)
 {
-    auto managed_callback_local = context->callback;
-    if (managed_callback_local)
+    auto managed_backgroud_error_callback_local = context->background_error_callback;
+    if (managed_backgroud_error_callback_local)
     {
-        managed_callback_local(status);
+        managed_backgroud_error_callback_local(status);
     }
 }
 
-API livemodel_context_t* CreateLiveModel(reinforcement_learning::utility::configuration* config)
-{
-    livemodel_context_t* context = new livemodel_context_t;
-    context->callback = nullptr;
-
-    context->livemodel = new reinforcement_learning::live_model(*config, pipe_managed_callback, context);
-
-    return context;
-}
-
-API void DeleteLiveModel(livemodel_context_t* context)
-{
-    // Since the livemodel destructor waits for queues to drain, this can have unhappy consequences,
-    // so detach the callback pipe first. This will cause all background callbacks to no-op in the
-    // unmanaged side, which maintains expected thread semantics (the user of the bindings)
-    context->callback = nullptr;
-
-    delete context->livemodel;
-    delete context;
+API livemodel_context_t* CreateLiveModel(reinforcement_learning::utility::configuration* config)
+{
+    livemodel_context_t* context = new livemodel_context_t;
+    context->background_error_callback = nullptr;
+    context->trace_logger_callback = nullptr;
+    context->trace_logger_factory = nullptr;
+
+    const auto binding_tracer_create = [context](reinforcement_learning::i_trace** retval,
+      const reinforcement_learning::utility::configuration& cfg,
+      reinforcement_learning::i_trace* trace_logger,
+      reinforcement_learning::api_status* status)
+    {
+      *retval = new reinforcement_learning::binding_tracer(context->trace_logger_callback);
+      return reinforcement_learning::error_code::success;
+    };
+
+    config->set(reinforcement_learning::name::TRACE_LOG_IMPLEMENTATION, reinforcement_learning::value::BINDING_TRACE_LOGGER);

```

Comment:
Nit: Since this constant is specific to C# binding,  should this constant only be inside the rl.net.native?

Reply:
Moved.